### DISCUSSION TITLE
Can we get affected rows in a query using driver.execute_rpc ?

### DISCUSSION THREAD

**Discussion by berusnet** [tdslite/tdslite#42] [Q&A] [answered]
Hello ! The answer seem no, so a question only to know which ca be a easy smart way

**Comment by xmkg** [comment, marked as answer]
Hello @berusnet,

Normally -- yes, you should be able to do so. The `execute_rpc` function returns a `tdsl::expected<tdsl::uint32_t, e_rpc_error_code>` object, which is a boolable object. If the object evaluates to `true`, then the object contains the amount of rows affected. Otherwise, it means execute_rpc has failed and the object contains the error code.

In theory, the code below illustrates how it should work:

```c++
 auto result = command_ctx.execute_rpc(tdsl::string_view{"INSERT INTO #test_rpc VALUES(0x0102030405)"}, params,
                            tdsl::detail::e_rpc_mode::executesql);                       
  if(result){
        auto rows_affected = result.get();
  } else {
      // Statement failed.
  }
```

... but it seems like the code has a slight bug that causes the code to process all done tokens instead of the ones with valid affected row count values, so the rows_affected will most probably show "0". I'll fix it with the next update.

**Reply by berusnet** [reply]
Thank you for your reply 

p.s yes result.get() show me always 0

**Comment by xmkg** [comment]
submitted a new issue for traceability: https://github.com/tdslite/tdslite/issues/43

**Reply by xmkg** [reply]
Hello,

This should be fixed with the 0.5.0 release that I just published. It may take some time for Arduino library index to pick up the new release. Meanwhile, you can download it manually from the releases page: https://github.com/tdslite/tdslite/releases/tag/v0.5.0

**Reply by berusnet** [reply]
Thank you, i have made two quick test and seem ok !

**Reply by xmkg** [reply]
You're welcome! Please let me know if you find any issues.
